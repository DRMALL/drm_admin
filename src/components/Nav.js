


import React, { Component } from 'react'
import Layout from 'antd/lib/layout'
import { siderArr } from '../commons/siderItem'
import { browserHistory } from 'react-router'
import dispatch from '../actions/dispatch'
import NavHead from './NavHead'
import whichSelect from '../utils/whichSelect'



const { Header, Content, Sider } = Layout
export default class Nav extends Component {
  componentDidMount() {
    // getMachineArr()
    // getMessageArr()
  }

  render(){
    const { sliderSelect } = this.props.state.nav
    return(
      <Layout className={(window.location.pathname==='/machine/edit') ?'nav-container' : 'nav-containers'}  >
       <Sider width={150} style={{ background: '#fff', borderRight: '1px solid #eee'}} >
         <div className='nav-sider-log' >Drm</div>
        {
          siderArr.map(item =>
            <div key={item.value} data-pathname={ item.path } onClick={ navToitem }
                 className='sider-item'  >
                <div className={ sliderSelect=== item.value ? `sider-icon nav-sider-${item.icon}-select` :`sider-icon nav-sider-${item.icon}`} >{item.value}</div>
            </div> )
        }
       </Sider>
        <Layout >
          <Header style={{ background: '#fff' }} >
            <NavHead  { ...this.props } />
          </Header>
           <Layout style={{ backgroundColor: '#F9FAFC', minWidth: 1000 }}  >
             <Content className='nav-content' >{
               this.props.children && React.cloneElement(this.props.children, { state: this.props.state, location: window.location, })
            }</Content>
           </Layout>
         </Layout>
      </Layout>
      )
  }
}

function  navToitem(e){
  let pathname  = e.currentTarget.dataset.pathname
  dispatch('NAV_ON_SELECT', whichSelect(pathname))
  browserHistory.push(pathname)
}




