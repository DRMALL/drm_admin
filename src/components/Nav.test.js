

import React from 'react'
import renderer from 'react-test-renderer'
import Nav from './Nav'
// import setUpfiles from '../setUpfiles'

it.skip('Nav snapshot test', ()=> {
  // afterEach(() => setUpfiles())

  const tree = renderer.create(<Nav />)
  expect(tree).toMatchSnapshot()

  // const wraaper = render( <Nav /> )
  // expect(toJson(wraaper)).toMatchSnapshot()
})
//此测试无效，主要是关于state的数据问题
