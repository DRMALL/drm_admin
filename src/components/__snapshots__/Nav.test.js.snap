// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav snapshot test 1`] = `
<div
  className="nav-containers ant-layout ant-layout-has-sider"
>
  <div
    className="ant-layout-sider"
    style={
      Object {
        "background": "#fff",
        "borderRight": "1px solid #eee",
        "flex": "0 0 150px",
        "width": "150px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <div
        className="nav-sider-log"
      >
        Drm
      </div>
      <div
        className="sider-item"
        data-pathname="/message"
        onClick={[Function]}
      >
        <div
          className="sider-icon nav-sider-message"
        >
          信息发布
        </div>
      </div>
      <div
        className="sider-item"
        data-pathname="/machine"
        onClick={[Function]}
      >
        <div
          className="sider-icon nav-sider-machine"
        >
          设备管理
        </div>
      </div>
      <div
        className="sider-item"
        data-pathname="/trouble"
        onClick={[Function]}
      >
        <div
          className="sider-icon nav-sider-trouble"
        >
          故障诊断
        </div>
      </div>
      <div
        className="sider-item"
        data-pathname="/parts"
        onClick={[Function]}
      >
        <div
          className="sider-icon nav-sider-parts"
        >
          配件管理
        </div>
      </div>
      <div
        className="sider-item"
        data-pathname="/user"
        onClick={[Function]}
      >
        <div
          className="sider-icon nav-sider-user"
        >
          用户管理
        </div>
      </div>
      <div
        className="sider-item"
        data-pathname="/authority"
        onClick={[Function]}
      >
        <div
          className="sider-icon nav-sider-authority-select"
        >
          权限管理
        </div>
      </div>
      <div
        className="sider-item"
        data-pathname="/workOrder"
        onClick={[Function]}
      >
        <div
          className="sider-icon nav-sider-work-order"
        >
          工单管理
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-layout"
  >
    <div
      className="ant-layout-header"
      style={
        Object {
          "background": "#fff",
        }
      }
    >
      <div
        className="nav-head-container"
      >
        <div
          className="nav-head-left nav-head-block"
        >
          admin
        </div>
        <div
          className="nav-hed-right nav-head-block"
          onClick={[Function]}
        >
          退出
        </div>
      </div>
    </div>
    <div
      className="ant-layout"
      style={
        Object {
          "backgroundColor": "#F9FAFC",
          "minWidth": 1000,
        }
      }
    >
      <div
        className="nav-content ant-layout-content"
      />
    </div>
  </div>
</div>
`;
